File: backend/app/api/routes/adsb.py
"""Read access to persisted ADS-B aircraft sightings."""
from __future__ import annotations

from fastapi import APIRouter, Depends

from app.api.deps import get_current_user
from app.db.models.user import User
from app.schemas.common import ok
from app.services.adsb_aircraft import DEFAULT_AIRCRAFT_MINUTES, list_aircraft

router = APIRouter(prefix="/api/adsb", tags=["adsb"])


@router.get("/aircraft")
async def list_adsb_aircraft(
    receiver_id: str | None = None,
    minutes: int = DEFAULT_AIRCRAFT_MINUTES,
    _: User = Depends(get_current_user),
) -> dict:
    aircraft = await list_aircraft(receiver_id=receiver_id, minutes=minutes)
    return ok(
        [
            {
                "receiver_id": a.receiver_id,
                "icao": a.icao,
                "last_type_code": a.last_type_code,
                "message_count": a.message_count,
                "first_seen_at": a.first_seen_at.isoformat(),
                "last_seen_at": a.last_seen_at.isoformat(),
                "latitude": a.latitude,
                "longitude": a.longitude,
            }
            for a in aircraft
        ]
    )

File: frontend/src/api/adsb.ts
import { api } from "../lib/apiClient";

export interface AdsbAircraft {
  receiver_id: string;
  icao: string;
  last_type_code: number;
  message_count: number;
  first_seen_at: string;
  last_seen_at: string;
  latitude: number | null;
  longitude: number | null;
}

export const listAdsbAircraft = (receiverId?: string, minutes?: number) => {
  const params = new URLSearchParams();
  if (receiverId) params.set("receiver_id", receiverId);
  if (minutes) params.set("minutes", String(minutes));
  const query = params.toString();
  return api.get<AdsbAircraft[]>(`/api/adsb/aircraft${query ? `?${query}` : ""}`);
};
